spec/unit/coinbase/smart_contract_spec.rb

```diff
@@ -198,6 +198,1229 @@
     end
   end
 
+  describe '.read' do
+    subject(:result) do
+      described_class.read(
+        network: network,
+        contract_address: contract_address,
+        method: method_name,
+        abi: abi,
+        args: args
+      )
+    end
+
+    let(:contract_address) { '0x1234567890123456789012345678901234567890' }
+    let(:method_name) { method_name_for_context }
+    let(:args) { {} }
+    let(:abi) { nil }
+
+    describe 'API interaction' do
+      let(:method_name_for_context) { 'testMethod' }
+      let(:abi) do
+        [{
+          'type' => 'function',
+          'name' => 'testMethod',
+          'inputs' => [],
+          'outputs' => [{ 'type' => 'uint256' }],
+          'stateMutability' => 'pure'
+        }]
+      end
+
+      before do
+        allow(smart_contracts_api).to receive(:read_contract).and_return(
+          Coinbase::Client::SolidityValue.new({
+                                                'type' => 'uint256',
+                                                'value' => '0'
+                                              })
```

Comment:
```suggestion
          Coinbase::Client::SolidityValue.new(
            type: 'uint256',
            value: '0'
          )
```

This other syntax is valid, but is not standard as of ruby 1.9